### PR TITLE
Added Sum of N Natural Numbers

### LINKED ISSUE #1272: Sum of first N natural numbers in C, C++, Java, Python & Go

### PR DESCRIPTION
**Pull request by MastersAbh**
Fixes #1272 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added implementation for finding sum of first N natural numbers in 5 languages

#### Languages Used:
- C
- C++
- Python
- Java
- Go

#### Files Added:
- Sum_of_N_Natural_Numbers.c
- Sum_of_N_Natural_Numbers.cpp
- Sum_of_N_Natural_Numbers.go
- Sum_of_N_Natural_Numbers.java
- Sum_of_N_Natural_Numbers.py

### CODE REVIEW

File: Sum_of_N_Natural_Numbers/Sum_of_N_Natural_Numbers.c [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    the number input to us. We use the well known formula,
+    sum = n * (n + 1)/2 to compute the sum
+
+*/
+
+#include<stdio.h>
+
+int main() {
+    int num;
+    printf("Enter N : ");
```
Comment (SakshiShreya) [inline Sum_of_N_Natural_Numbers/Sum_of_N_Natural_Numbers.c]: Remove space before `:` in all the files.

Comment (MastersAbh) [reply Sum_of_N_Natural_Numbers/Sum_of_N_Natural_Numbers.c]: Done. @SakshiShreya Please review.

**Comment by SakshiShreya** [comment]
You can also write the message inside `input()` in python.

File: Sum_of_N_Natural_Numbers/Sum_of_N_Natural_Numbers.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+*/
+
+#include<bits/stdc++.h>
+using namespace std;
+
+int main() {
+    int num;
+    cout << "Enter N: ";
+    cin >> num;
```
Comment (marwahmanbir) [inline Sum_of_N_Natural_Numbers/Sum_of_N_Natural_Numbers.cpp]: There is no need to give any space in the cout & cin statements. The compiler just ignores unnecessary whitespaces. Approving this for now.

**Comment by marwahmanbir** [comment]
Good work :+1:

**Comment by jainaman224** [comment]
Can we have a better file name?

**Comment by marwahmanbir** [comment]
"Natural_Numbers_Sum" could be better I guess?

**Comment by MastersAbh** [comment]
Done. Please review @marwahmanbir

**Pull request merged**